grakn-core/src/main/java/ai/grakn/graql/Pattern.java

```diff
@@ -42,4 +42,16 @@
      */
     @CheckReturnValue
     PatternAdmin admin();
+
+    /**
+     * Join patterns in a conjunction
+     */
+    @CheckReturnValue
+    Pattern and(Pattern pattern);
```

Comment:
This is a helpful sugar method that proved useful for writing the Quickcheck properties (and useful for normal users too). It allows you to write `and` and `or` like actual binary operators 

e.g. we have tests stating `match(patternA.and(patternB)) == match(patternB.and(patternA))`